soh/soh/UIWidgets.hpp

```diff
@@ -22,6 +22,15 @@ namespace UIWidgets {
             }
             return 1;
         }
+
+        static int FilterAlphaNum(ImGuiInputTextCallbackData* data) {
+            const char* alphanum = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWYZ0123456789";
+            if (data->EventChar < 256 && strchr(alphanum, (char)data->EventChar)) {
+                return 0;
+            }
+            return 1;
+        }
```

Comment:
we're in cpp here, we can use `bool` and `return true`/`return false`

Reply:
I was following similar structure from Archez's `FilterNumbers`, but will apply.

Reply:
This is a callback for TextCharFilterCallback on imgui side which is supposed to return an int. Where 0 means keep the char, and anything else means discard the char

Reply:
It'd probably be best to leave it alone if that's the case then.